src/box/box.cc

```diff
@@ -1111,6 +1115,18 @@ box_check_small_alloc_options(void)
 			  " to 1024 * 16 and exponent of two");
 }
 
+static void
+box_check_iproto_options(void)
+{
+	int iproto_threads = cfg_geti("iproto_threads");
+	if (iproto_threads <= 0 || iproto_threads > IPROTO_THREADS_MAX) {
+		tnt_raise(ClientError, ER_CFG, "iproto_threads",
```

Comment:
In the new code we use diag_set() and return 0/-1. Exceptions are getting deleted. Please, don't introduce new ones.

Reply:
I change this to diag_set, but in `box_check_config`, i should to call `diag_raise`, which throw exception, because
`box_check_confg` used in lua/cfg.cc   in this way:
```c++
static int
lbox_cfg_check(struct lua_State *L)
{
        try {
                box_check_config();
        } catch (Exception *) {
                luaT_error(L);
        }
        return 0;
}
```